scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/utils/TaskKiller.java

```diff
@@ -54,13 +54,28 @@ public synchronized boolean wasKilled() {
         return wasKilled;
     }
 
-    public synchronized void kill(Status status) {
-        wasKilled = true;
-        this.status = status;
-        threadToKill.interrupt();
+    /**
+     * Interrupts the target thread repeatedly as InterruptedException can be catched without action
+     */
+    private void interruptRepeatedly() {
+        while (threadToKill.isAlive()) {
+            threadToKill.interrupt();
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)